Tests/TuistCacheTests/Mappers/Graph/TreeShakePrunedTargetsGraphMapperTests.swift

```diff
@@ -241,4 +241,36 @@ final class TreeShakePrunedTargetsGraphMapperTests: TuistUnitTestCase {
         // Then
         XCTAssertEqual(gotGraph, expectedGraph)
     }
+
+    func test_map_preserves_target_order_in_projects() throws {
+        // Given
+        let firstTarget = Target.test(name: "Brazil")
+        let secondTarget = Target.test(name: "Ghana")
+        let thirdTarget = Target.test(name: "Japan")
+        let prunedTarget = Target.test(name: "Pruned", prune: true)
+        let project = Project.test(targets: [firstTarget, secondTarget, thirdTarget, prunedTarget])
+
+        let graph = Graph.test(
+            path: project.path,
+            projects: [project.path: project],
+            targets: [project.path: [
+                firstTarget.name: firstTarget,
+                secondTarget.name: secondTarget,
+                thirdTarget.name: thirdTarget,
+                prunedTarget.name: prunedTarget
+            ]],
+            dependencies: [:]
+        )
+
+        let expectedProject = Project.test(targets: [firstTarget, secondTarget, thirdTarget])
+
+        // When
+        let (gotGraph, _) = try subject.map(graph: graph)
+
+        print(graph.projects.first!.value.targets.map(\.name))
+        print(gotGraph.projects.first!.value.targets.map(\.name))
```

Comment:
Shouldn't this XCTAssert on it rather than printing? :D

Reply:
If we are making sure that order of targets is the same, does it make sense to zip both sequences and just compare them?

i.e, 

```
for (generatedTarget, expectedTarget) in zip(graph.projects.first!.value.targets, expectedProject.targets) {
     XCTAssertEqual(generatedTarget expectedTarget)
}
```

Reply:
Ahh I forgot them in there.

I think the `XCTAssertEqual(gotGraph.projects.first?.value, expectedProject)` is enough to cover the order of projects based on the `Equatable` conformance of `Project`. Let me know if you think the assertion should be more specific.

Reply:
You are right, I didn't consider that comparing the projects we were also comparing the targets 😁